tests/test_video.py

```diff
@@ -282,3 +290,432 @@ def test_make_video(self, reset_db: None) -> None:
+        if len(segments) > 0:
+            validation_t = pxt.create_table('segment_validation', {'segment': pxt.Video}, media_validation='on_write')
+            validation_t.insert([{'segment': row['url']} for row in segments], on_error='abort')
+            pxt.drop_table('segment_validation')
+
+    @pytest.mark.parametrize('segment_duration', [5.0, 10.0, 100.0])
+    def test_video_splitter(self, segment_duration: float, reset_db: None) -> None:
+        skip_test_if_not_in_path('ffmpeg')
+        from pixeltable.iterators.video import VideoSplitter
+
+        video_filepaths = get_video_files()
+        for min_segment_duration in [0.0, segment_duration]:
+            for overlap in [0.0, 1.0, 4.0]:
+                t = pxt.create_table('videos', {'video': pxt.Video})
+                t.insert([{'video': p} for p in video_filepaths])
+                s = pxt.create_view(
+                    'segments_5s',
+                    t,
+                    iterator=VideoSplitter.create(
+                        video=t.video,
+                        segment_duration=segment_duration,
+                        overlap=overlap,
+                        min_segment_duration=min_segment_duration,
+                    ),
+                )
+                self._validate_splitter_segments(t, s, overlap, min_segment_duration)
+                pxt.drop_table('videos', force=True)
+
+    def test_video_splitter_errors(self, reset_db: None) -> None:
+        skip_test_if_not_in_path('ffmpeg')
+
+        from pixeltable.iterators.video import VideoSplitter
+
+        t = pxt.create_table('videos', {'video': pxt.Video})
+        with pytest.raises(pxt.Error, match='segment_duration must be a positive number'):
+            _ = pxt.create_view('s', t, iterator=VideoSplitter.create(video=t.video, segment_duration=-1))
+
+        with pytest.raises(pxt.Error, match='overlap must be less than segment_duration'):
+            _ = pxt.create_view('s', t, iterator=VideoSplitter.create(video=t.video, segment_duration=1, overlap=1))
+
+        with pytest.raises(pxt.Error, match='segment_duration must be at least min_segment_duration'):
+            _ = pxt.create_view(
+                's', t, iterator=VideoSplitter.create(video=t.video, segment_duration=1, min_segment_duration=2)
+            )
+
+    def test_overlay_text(self, reset_db: None, tmp_path: Path) -> None:
+        skip_test_if_not_in_path('ffmpeg')
+
+        t = pxt.create_table('videos', {'video': pxt.Video})
+        t.add_computed_column(clip_5s=t.video.clip(start_time=0, duration=5))
+
+        text = "Line 1\nLine2: 'quoted text'"
+        t.add_computed_column(
+            o1=t.clip_5s.overlay_text(
+                text,
+                color='black',
+                opacity=0.5,
+                horizontal_align='left',
+                horizontal_margin=10,
+                vertical_align='top',
+                box=False,
+            )
+        )
+        t.add_computed_column(
```

Comment:
It's hard to test the output because the output is a video.

But we COULD break up the logic in `overlay_text()` via a `__overlay_params_to_drawtext()` subroutine, and test that thing, which would at least make sure that we're correctly mapping various combinations of "our" parameters onto the expected ffmpeg inputs.

Reply:
I'm not sure I agree with this. It's a large set of method parameters that interact in complicated ways and whose behavior currently isn't tested at all. That's just asking for something to break the next time someone refactors or adds a new parameter.

It also wouldn't be a lot of time, say 10 minutes to do the refactoring and 30 minutes to add a decent set of test cases. Passing up an opportunity to improve our test surface in order to save ~45 minutes doesn't seem like a good tradeoff to me?